main.go

```diff
@@ -26,9 +26,16 @@ func main() {
 func realMain() (returnCode int) {
 	l := configureLogging("hcdiag")
 
+	// Ensure the first arg is a flag
+	err := noSubcommand(os.Args)
+	if err != nil {
+		l.Error("subcommands not supported", "error", err)
+		os.Exit(2)
```

Comment:
Since we're calling this from `main`, which does an `os.Exit`, should we just return 2 here?

Reply:
Oh - yes!